firebase-sessions/src/main/kotlin/com/google/firebase/sessions/ProcessUtils.kt

```diff
@@ -0,0 +1,84 @@
+/*
+ * Copyright 2023 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.google.firebase.sessions
+
+import android.app.ActivityManager
+import android.app.ActivityManager.RunningAppProcessInfo
+import android.app.Application
+import android.content.Context
+import android.os.Build
+import android.os.Process
+
+/** Utils related to processes. */
+internal object ProcessUtils {
+  /** Returns whether the current process should generate a new session or not. */
+  fun shouldProcessGenerateNewSession(context: Context): Boolean =
+    isDefaultProcess(context) && isForegroundImportance()
+
+  /** Returns whether the current process is the app's default process or not. */
+  private fun isDefaultProcess(context: Context): Boolean =
+    getCurrentProcessName(context) == getDefaultProcessName(context)
+
+  /** Returns whether the current process or service is running in the foreground or not. */
+  private fun isForegroundImportance(): Boolean {
```

Comment:
I'm not sure what this function is supposed to return. Are we checking if the current process is in Foreground?